packages/protocol/contracts/governance/Validators.sol

```diff
@@ -58,16 +61,36 @@ contract Validators is IValidators, Ownable, ReentrancyGuard, Initializable, Usi
   struct ValidatorGroup {
     string name;
     string url;
+    LinkedList.List members;
     // TODO(asa): Add a function that allows groups to update their commission.
     FixidityLib.Fraction commission;
-    LinkedList.List members;
+  }
+
+  // Stores the epoch number at which a validator joined a particular group.
+  struct MembershipHistoryEntry {
+    uint256 epochNumber;
+    address group;
+  }
+
+  // A circular buffer storing the membership history of a validator.
+  struct MembershipHistory {
```

Comment:
We do this validator history because of the authorized keys. But, can't we use the "account" address as a validator identifier somewhere?
It looks like that the validator's `identity` is the account address, not the current authorized address.

Reply:
btw, what happens if the authorize address for a validator changes during a epoch in which that validator is within the validatorSet?

Reply:
Correction, this comment is not for MembershipHistory, but for what we do in `LockedGold`; where we save all keys that at some point where part of the validators authorized keys.